llm/gguf.go

```diff
@@ -199,21 +209,84 @@ func (llm *ggufModel) Decode(r io.Reader) error {
 
 	// decode tensors
 	for i := 0; uint64(i) < llm.NumTensor(); i++ {
-		if _, err := llm.readString(r); err != nil {
+		name, err := llm.readString(ro)
+		if err != nil {
 			return err
 		}
 
-		dimensions := llm.readU32(r)
+		dims := llm.readU32(ro)
```

Comment:
this, and filling in `nes` probably deserves a comment.

Reply:
is it possible for dims to be larger than 4?

Reply:
not that I know of. it's hard coded to 4 in llama.cpp